File: src/Unit-3/DoThis/GithubClientFactory.cs
using Octokit;
using Octokit.Internal;

namespace GithubActors
{
    /// <summary>
    /// Creates <see cref="GitHubClient"/> instances.
    /// </summary>
    public static class GithubClientFactory
    {
        /// <summary>
        /// OAuth token - necessary to generate authenticated requests
        /// and achieve non-terrible hourly API rate limit
        /// </summary>
        public static string OAuthToken { get; set; }

        public static GitHubClient GetUnauthenticatedClient()
        {
            return new GitHubClient(new ProductHeaderValue("AkkaBootcamp-Unit3"));
        }

    public static GitHubClient GetClient()
        {
            return new GitHubClient(new ProductHeaderValue("AkkaBootcamp-Unit3"), new InMemoryCredentialStore(new Credentials(OAuthToken)));
        }
    }
}

File: src/Unit-3/DoThis/GithubClientFactory.fs
namespace GithubActors

module GithubClientFactory =

    open Octokit
    open Octokit.Internal

    let mutable oauthToken = ""

    let setOauthToken token = oauthToken <- token

    let getUnauthenticatedClient () =
        GitHubClient (ProductHeaderValue "AkkaBootcamp-Unit3")

    let getClient () =
         GitHubClient (ProductHeaderValue "AkkaBootcamp-Unit3", InMemoryCredentialStore (Credentials oauthToken))